subsys/bluetooth/controller/ll_sw/ull.c

```diff
@@ -393,48 +391,56 @@ u8_t ll_rx_get(void **node_rx, u16_t *handle)
 	*node_rx = NULL;
 
 	link = memq_peek(memq_ll_rx.head, memq_ll_rx.tail, (void **)&rx);
-	if (link) {
-#if defined(CONFIG_BT_CONN)
-		cmplt = tx_cmplt_get(handle, &mfifo_tx_ack.f, rx->ack_last);
-		if (!cmplt) {
-			u8_t f, cmplt_prev, cmplt_curr;
-			u16_t h;
-
-			cmplt_curr = 0U;
-			f = mfifo_tx_ack.f;
-			do {
-				cmplt_prev = cmplt_curr;
-				cmplt_curr = tx_cmplt_get(&h, &f,
-							  mfifo_tx_ack.l);
-			} while ((cmplt_prev != 0U) ||
-				 (cmplt_prev != cmplt_curr));
-
-			/* Do not send up buffers to Host thread that are
-			 * marked for release
-			 */
-			if (rx->type == NODE_RX_TYPE_DC_PDU_RELEASE) {
-				(void)memq_dequeue(memq_ll_rx.tail,
-						   &memq_ll_rx.head, NULL);
-				mem_release(link, &mem_link_rx.free);
-
-				LL_ASSERT(mem_link_rx.quota_pdu < RX_CNT);
-				mem_link_rx.quota_pdu++;
-
-				mem_release(rx, &mem_pdu_rx.free);
-
-				rx_alloc(1);
-
-				goto ll_rx_get_again;
-			}
-#endif /* CONFIG_BT_CONN */
-
-			*node_rx = rx;
+	if (IS_ENABLED(CONFIG_BT_CONN)) {
```

Comment:
General comment: 
Pro: I think the readability is increased (slightly) with this `IS_ENABLED`. Nice.
Con: As pointed was out before, indentation is increased by 1 level. To avoid this extra level of indent, I prefer to have the code refactored so big blocks of typical-case stuff are not nested. Example:

Change from:
```
if (typical happy-day condition: link or IS_ENABLED(foo) etc) {
    long body, that now has gained 1 more indent level.
    long body
    long body
    long body
    long body
    long body
} else {
    error handling
}
```

to:
```
if (typical happy-day condition violated) {
    error handling
    return;
} 

long body, that now retains its indent level
long body
long body
long body
long body
long body

```

This style the added benefits of being more linear and has a more Error-handling first perspective.
Granted, MISRA may not like multiple returns :/